src/payment/strategies/zip/zip-payment-strategy.spec.ts

```diff
@@ -0,0 +1,229 @@
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+                .mockReturnValue(undefined);
+            try {
+                await strategy.initialize(zipOptions);
+            } catch (error) {
+                expect(error).toBeInstanceOf(MissingDataError);
+                expect(zipScriptLoader.load).not.toHaveBeenCalled();
+
+            }
+        });
+    });
+
+    describe('#execute()', () => {
+        let zipOptions: PaymentInitializeOptions;
+        let orderRequestBody: OrderRequestBody;
+
+        beforeEach(async () => {
+            orderRequestBody = getOrderRequestBody();
+            zipOptions = { methodId: 'zip' };
+
+            await strategy.initialize(zipOptions);
+        });
+
+        it('dispatch widget interaction when execute is called', async () => {
+            await strategy.execute(orderRequestBody, zipOptions);
+
+            expect(store.dispatch).toHaveBeenCalled();
+        });
+
+        it('expect to get the payment method', async () => {
+            await strategy.execute(orderRequestBody, zipOptions);
+
+            expect(store.getState().paymentMethods.getPaymentMethod).toHaveBeenCalledWith(zipOptions.methodId);
+        });
+
+        it('returns the payment information', async () => {
+            await strategy.execute(orderRequestBody, zipOptions);
+
+            expect(store.getState).toHaveBeenCalled();
+        });
+
+        it('succesfully dispatches widget Interaction', async () => {
+            await strategy.execute(orderRequestBody, zipOptions);
+
+            expect(paymentMethodActionCreator.loadPaymentMethod).toHaveBeenCalledWith('zip');
+            expect(store.getState().paymentMethods.getPaymentMethod).toHaveBeenCalledWith('zip');
+        });
+
+        it('inits the checkout of zipClient with the expected payload', async () => {
+            const expectedPayload =  {
+                onComplete: expect.any(Function),
+                onCheckout: expect.any(Function),
+            };
+            await strategy.execute(orderRequestBody, zipOptions);
+
+            expect(zipClient.Checkout.init).toHaveBeenCalledWith(expectedPayload);
+        });
+
+        it('completes the checkout and submit the order', async () => {
+            await strategy.execute(orderRequestBody, zipOptions);
+
+            expect(orderActionCreator.submitOrder).toBeCalled();
+            expect(paymentActionCreator.submitPayment).toBeCalled();
+        });
+
+        it('cancels the checkout if the lightbox is closed', async () => {
+            zipClient = getZipScriptMock(false);
+            jest.spyOn(zipScriptLoader, 'load')
+                .mockReturnValueOnce(Promise.resolve(zipClient));
+            await strategy.initialize(zipOptions);
+            await strategy.execute(orderRequestBody, zipOptions);
+
+            expect(orderActionCreator.submitOrder).not.toBeCalled();
+            expect(paymentActionCreator.submitPayment).not.toBeCalled();
```

Comment:
Should probably also verify that `PaymentMethodCancelledError` is returned / thrown.